ansible-completion.bash

```diff
@@ -30,10 +30,17 @@ _ansible_complete_host() {
     local last_word=${current_word##*:}
     local grep_opts="-o"
 
-    # The 'tail' call is there to get rid of the first 'hosts (xx)' line in the
-    # output.
-    local hosts_and_groups=$(ansible all --list-hosts 2> /dev/null | tail +2 &&
-        ansible-inventory --list | jq -r "keys | .[]")
+    if [[ ! -f "$HOME/.cache/ansible-completion/ansible_inventory_content.txt" ]]; then
+        mkdir -p $HOME/.cache/ansible-completion
+
+        # The 'tail' call is there to get rid of the first 'hosts (xx)' line in
+        # the output. 'awk' call is there to trim leading and trailing
+        # whitespace from the output.
+        ansible all --list-hosts 2> /dev/null | tail +2 | awk '{$1=$1};1' > $HOME/.cache/ansible-completion/ansible_inventory_content.txt
+        ansible-inventory --list | jq -r "keys | .[]" >> $HOME/.cache/ansible-completion/ansible_inventory_content.txt
```

Comment:
How does this work with multiple Ansible repositories on the same host? :thinking:

Reply:
Poorly, likely. :unamused: It doesn't really scale beyond a single `ansible.cfg` AFAICT. I wonder how we could make it be more generally usable in that sense. One way would be to be able to override the `$HOME/.cache/ansible-completion`-path using an ENV variable or something. Any other ideas?

Reply:
Add `$CWD` to the name of the file in `.cache`?

Reply:
Hmm, `$CWD` doesn't seem to be defined in my `bash` at least. Are you referring to the full `pwd` or something like `basename $(pwd)`?

Reply:
Yeah, the full `$(pwd)` or something like that. Slap a `md5sum` around it to make it safe for file name usage?

Reply:
Thanks. I added an implementation now (had to pipe it through `awk` to get rid of some extra noise that `md5um` prints out). With these changes in place, it creates the file like this:

```
$ tree ~/.cache/ansible-completion/
/home/per/.cache/ansible-completion/
└── 50b28211a61d16b65ceb2977e2f8a2a9
    └── ansible_inventory_content.txt

1 directory, 1 file
```

Reply:
@slovdahl Anything holding this back from approval now?

Reply:
No, I didn't realize you wanted my approval :joy: